### PR TITLE
gh-120111: Don't use cirrus M1 macOS runners on fork

### PR DESCRIPTION
**Pull request by nineteendo**
See https://github.com/nineteendo/cpython/pull/5 for how this works on a fork.

* Issue: gh-120111

### CODE REVIEW

**Comment by nineteendo** [comment]
@hugovk, we now use macOS-14 instead of cirrus when running on a fork.

File: .github/workflows/reusable-macos.yml [hunk 1]
```diff
@@ -27,6 +27,13 @@ jobs:
       fail-fast: false
       matrix:
         os: ${{fromJson(inputs.os-matrix)}}
+        isFork:
```
Comment (hugovk) [inline .github/workflows/reusable-macos.yml]: We tend to use kebab-case elsewhere:
```suggestion
        is-fork:
```

File: .github/workflows/reusable-macos.yml [hunk 2]
```diff
@@ -27,6 +27,13 @@ jobs:
...
+          - ${{ github.repository_owner != 'python' }}
+        exclude:
+          - os: "ghcr.io/cirruslabs/macos-runner:sonoma"
+            isFork: true
```
Comment (hugovk) [inline .github/workflows/reusable-macos.yml]:
```suggestion
            is-fork: true
```

File: .github/workflows/reusable-macos.yml [hunk 3]
```diff
@@ -27,6 +27,13 @@ jobs:
...
+          - os: "macos-14"
+            isFork: false
```
Comment (hugovk) [inline .github/workflows/reusable-macos.yml]:
```suggestion
            is-fork: false
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -211,7 +211,8 @@ jobs:
       config_hash: ${{ needs.check_source.outputs.config_hash }}
       free-threading: true
       # Cirrus is M1
```
Comment (hugovk) [inline .github/workflows/build.yml]:
```suggestion
      # Cirrus and macos-14 are M1.
```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -211,7 +211,8 @@ jobs:
...
-      os-matrix: '["ghcr.io/cirruslabs/macos-runner:sonoma"]'
+      # macos-14-large is Intel with 12 cores (most parallelism)
```
Comment (hugovk) [inline .github/workflows/build.yml]:
```suggestion
      # Cirrus used for upstream, macos-14 for forks.
```

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -199,8 +199,8 @@ jobs:
     uses: ./.github/workflows/reusable-macos.yml
     with:
       config_hash: ${{ needs.check_source.outputs.config_hash }}
-      # Cirrus is M1, macos-13 is default GHA Intel
-      os-matrix: '["ghcr.io/cirruslabs/macos-runner:sonoma", "macos-13"]'
+      # Cirrus and macos-14 are M1, macos-13 is default GHA Intel
```
Comment (hugovk) [inline .github/workflows/build.yml]:
```suggestion
      # Cirrus and macos-14 are M1, macos-13 is default GHA Intel.
      # Cirrus used for upstream, macos-14 for forks.
```

**Comment by nineteendo** [comment]
I applied your suggestions.

**Review by hugovk — APPROVED** [review]
Thank you!

**Pull request merged**